src/pages/personalisation/BaseMultiQuestion.js

```diff
@@ -118,11 +121,30 @@ class BaseMultiQuestion extends BaseQuestion {
         storage.setJSON(this.props.name, Array.from(this.selected));
     }
 
+    shouldRenderFloatingDoneButton(): string {
+        if (this.state.windowHeight < this.state.rootHeight - 60) {
+            return "hasFloatingDoneButton";
+        }
+        return "";
+    }
+
+    componentDidMount(): void {
+        this.setState({windowHeight: window.innerHeight});
+        this.setState({rootHeight: ReactDOM.findDOMNode(this).offsetHeight});
+    }
+
     render(): ReactElement {
         let selected = this.selected;
 
         return (
-            <div className="BaseMultiQuestion">
+            <div
+                className={
+                    classnames(
+                        "BaseMultiQuestion",
+                        this.shouldRenderFloatingDoneButton()
```

Comment:
`classnames` can take a class name and a truthy statement to evaluate: https://github.com/JedWatson/classnames